test/cpp/tensorexpr/test_kernel.cpp

```diff
@@ -66,7 +77,17 @@ void testKernel_2() {
   TensorExprKernel k(graph);
   std::vector<at::Tensor> inputs = {a, b};
   Stmt* s = k.getCodeGenStmt();
-  // TODO: verify stmt
+
+  std::ostringstream oss;
+  oss << *s;
+
+  // Check the IR we produced
+  const std::string& verification_pattern =
+      R"IR(
+# CHECK: for (int v = 0; v < 5; v++)
+# CHECK-NEXT: for (int v_1 = 0; v_1 < 3; v_1++)
+# CHECK-NEXT: aten_mul[v_1 + 3 * v] = (t0[v_1 + 3 * v]) * ((t0[v_1 + 3 * v]) * (t1[5 * v_1 + v])))IR";
```

Comment:
I think this check could be too brittle (echoing what you said earlier). Would it be possible to not check things that we don't plan to keep stable, like variable names and unnecessary braces in the expressions?

That said, I think it could be useful to have an exact check in one place just to make sure we're not missing changes. It will not be a very high burden to update this one place when a change does happen. What do you think?

Reply:
I went back and forth on this. The actual index formula is the valuable part; if I remove it, the test just doesn't have all that much value. Variable names themselves are relevant too, they test we're using the right permutation of inputs.

An alternative would be to use captures to parametrize it a bit, but that looks particularly gnarly and I'm no file check expert either. If this proves to be a burden, we'll revisit.

Reply:
I'm forced to reconsider given the test failures with the LLVM backend.

Reply:
Done.